### PR TITLE
Update generate.rb to create json files for casks in new directories for API

### PR DESCRIPTION
**Pull request by cmbernard333**
#13 

Add json generation to `generate.rb` for casks.

This does **not** generate analytics or HTML as this will be addressed in a separate PR.

### CODE REVIEW

File: script/generate.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env brew ruby
-directories = ["_data/formula", "api/formula", "formula"]
+require 'cask/cask'
```
Comment (MikeMcQuaid) [inline script/generate.rb]: Maybe move the require directory about the `Cask::Cask.each`?

File: script/generate.rb [hunk 2]
```diff
@@ -12,3 +13,14 @@
   html.gsub!("title: $TITLE", "title: #{f.name}")
   IO.write("formula/#{f.name}.html", html)
 end
+
+Cask::Cask.each do |c|
+  json_filename = "#{File.basename(c.sourcefile_path,'.rb')}.json"
+  IO.write("_data/cask/#{json_filename}", JSON.pretty_generate(c.to_h))
+  FileUtils.cp "_api_formula.json.in", "api/cask/#{json_filename}"
```
Comment (MikeMcQuaid) [inline script/generate.rb]: Does this file work perfectly? If so, can it be renamed to `_api.json.in` instead? Thanks!

File: script/generate.rb [hunk 3]
```diff
@@ -12,3 +13,14 @@
...
+
+  # TODO: generate cask html template and output(s)
+  # html = IO.read "_formula.html.in"
+  # html.gsub!("title: $TITLE", "title: #{c.name}")
+  # IO.write("cask/#{c.name}.html", html)
```
Comment (MikeMcQuaid) [inline script/generate.rb]: Can remove the commented-out, non-TODO lines.

File: script/generate.rb (hunk 2 above)
Comment (cmbernard333) [reply script/generate.rb]: Yes it does work (see below). Should this also be adjusted for formula?

```
christianbernard@Dirac-MBP formulae.brew.sh (feature/formulae.brew.sh-generate-cask-json) $ cat _data/cask/chromium.json
{
  "name": [
    "Chromium"
  ],
  "homepage": "https://www.chromium.org/Home",
  "url": "https://download-chromium.appspot.com/dl/Mac?type=snapshots",
  "appcast": null,
  "version": "latest",
  "sha256": "no_check",
  "artifacts": [
    [
      "chrome-mac/Chromium.app"
    ],
    {
      "trash": [
        "~/Library/Preferences/org.chromium.Chromium.plist",
        "~/Library/Caches/Chromium",
        "~/Library/Application Support/Chromium",
        "~/Library/Saved Application State/org.chromium.Chromium.savedState"
      ],
      "signal": [

      ]
    }
  ],
  "caveats": "",
  "depends_on": {
  },
  "conflicts_with": {
  },
  "container": null,
  "auto_updates": null
```

File: script/generate.rb (hunk 1 above)
Comment (cmbernard333) [reply script/generate.rb]: Will do.

File: script/generate.rb (hunk 2 above)
Comment (MikeMcQuaid) [reply script/generate.rb]: Seems good 👍

**Comment by MikeMcQuaid** [comment]
Thanks @cmbernard333, nice work! Should generate this on the next homebrew-core push. Will need to add a similar thing to https://github.com/Homebrew/homebrew-core/blob/master/.travis.yml to homebrew-cask

**Comment by cmbernard333** [comment]
@MikeMcQuaid Are we waiting on analytics as well (PR #92) so we can do analytics generation after creating the JSON for the cask API? Or should I just create a PR to call the script in the .travis.yml inside homebrew-cask?

**Comment by MikeMcQuaid** [comment]
@cmbernard333 The analytics are done every 24h rather than on every push. Feel free to pick anything up that's in progress and cherry-pick commits. Thanks!

**Comment by MikeMcQuaid** [comment]
@cmbernard333 These are not being generated: https://github.com/Homebrew/formulae.brew.sh/tree/master/_data

**Comment by cmbernard333** [comment]
@MikeMcQuaid per which build? The homebrew-cask build?

**Comment by MikeMcQuaid** [comment]
@cmbernard333 my expectation from this PR was that the generate script would start committing those files but this has not happened (see https://github.com/Homebrew/formulae.brew.sh/tree/master/_data). Perhaps they need to be manually committed first.

**Comment by cmbernard333** [comment]
That makes sense. Should I create a PR for this repo with the generated homebrew casks based on the latest set of casks?

**Comment by cmbernard333** [comment]
Here is a PR adding the initial set of json for casks in `_data/cask` - this does not include the HTML yet

**Comment by MikeMcQuaid** [comment]
I'm now thinking it may be worth having a separate generate script that's run for formulae and for casks as otherwise we'll need to make sure both repos are up to date to generate either. Separate jobs mean we can just run one of them from each relevant CI (homebrew-core and homebrew-cask).

**Comment by cmbernard333** [comment]
@MikeMcQuaid Roger. I'll make a separate PR and make a new script `generate-cask.rb`.

**Pull request merged**